src/components/Form/singleValueForm.tsx

```diff
@@ -0,0 +1,141 @@
+import { useWeb3Modal } from '@web3modal/react';
+import { ContractInterface, ethers } from 'ethers';
+import { Field, Form, Formik } from 'formik';
+import { useProvider, useSigner } from 'wagmi';
+import * as Yup from 'yup';
+import { createMultiStepsTransactionToast, showErrorTransactionToast } from '../../utils/toast';
+import Loading from '../Loading';
+import SubmitButton from './SubmitButton';
+import { useChainId } from '../../hooks/useChainId';
+import { ObjectShape } from 'yup/lib/object';
+import { FEE_RATE_DIVIDER } from '../../config';
+
+interface IFormValuesNumber {
+  value?: number;
+}
+interface IFormValuesString {
+  value?: string;
+}
+
+interface validationDatasType {
+  valueType: string;
+  validationSchema?: Yup.ObjectSchema<ObjectShape>;
+  initialValue?: number | string;
+  selectOptions?: { value: string; label: string }[];
+  shouldMultiplyByFeeRate?: boolean;
+}
+
+interface contractParamsType {
+  contractFunctionName: string;
+  contractEntity: string;
+  contractInputs: unknown;
+  contractAddress: string;
+  contractAbi: ContractInterface;
+}
+
+function SingleValueForm({
+  validationDatas,
+  contractParams,
+  valueName,
+}: {
+  validationDatas: validationDatasType;
+  contractParams: contractParamsType;
+  valueName: string;
+}) {
+  const { validationSchema, valueType, initialValue, selectOptions, shouldMultiplyByFeeRate } =
+    validationDatas;
+  const { contractFunctionName, contractEntity, contractInputs, contractAddress, contractAbi } =
+    contractParams;
+
+  const chainId = useChainId();
+  const { open: openConnectModal } = useWeb3Modal();
+  const provider = useProvider({ chainId });
+  const { data: signer } = useSigner({
+    chainId,
+  });
+  let contract: ethers.Contract;
+
+  if (!signer) {
+    return <Loading />;
+  }
+
+  if (signer) {
+    contract = new ethers.Contract(contractAddress, contractAbi, signer);
+  }
+
+  const onSubmit = async (
```

Comment:
- file need to me SingleValueForm 
- I think that's better to split this abstraction and have SingleFeeRateForm and SingleFeeAmountForm